tests/installation/agama_reboot.pm

```diff
@@ -38,9 +41,65 @@ sub get_agama_install_console_tty {
     return 7;
 }
 
+sub verify_agama_auto_install_done_cmdline {
+    # for some remote workers, there is no vnc access to the install console,
+    # so we need to make sure the installation has completed from command line.
+    my $timeout = 300;
+    while ($timeout > 0) {
+        if (script_run("journalctl -u agama | grep 'Install phase done'") == 0) {
+            record_info("agama install phase done");
+            return;
+        }
+        sleep 20;
+        $timeout = $timeout - 20;
+    }
+    die "Install phase is not done, please check agama logs";
+}
+
+sub agama_system_prepare_ipmi {
+    # Mount the root disk after installation, and do some prepare tasks:
+    # Configure serial console, enable root ssh access, etc
+
+    # Get the root disk name
+    my $device = script_output qq(agama config show | grep -oP '(?<="disk": ")[^"]+');
+    # Use partition prefix for nvme devices
+    my $prefix = $device =~ /nvme/ ? "p" : "";
+    my $root_partition_id = 2;
+    my $root_partition = "${device}" . $prefix . $root_partition_id;
+    record_info("Device information", "Device: ${device}\nRoot partition: ${root_partition}");
+    assert_script_run("mount ${root_partition} /mnt");
+    # Set correct serial console to be able to see login in first boot
+    my $sol_console = get_required_var('IPXE_CONSOLE');
```

Comment:
This won't work for all ipmi machines. Not every ipmi machine has such setting, eg https://openqa.suse.de/tests/16174750/file/vars.json. Actually virtualization squad's ipmi machines use `SERIALDEV`.

Reply:
fixed, yes, not all ipmi setups use ipxe boot. thanks.